### PR TITLE
Make logits in cudadecoder const.

### PR DESCRIPTION
**Pull request by galv**
There is no reason why they couldn't be const. It was just an
oversight in the interface.

### CODE REVIEW

**Comment by kkm000** [comment]
LGTM, thanks, merging!
